src/material/button-toggle/button-toggle.ts

```diff
@@ -437,8 +511,16 @@ export class MatButtonToggle implements OnInit, AfterViewInit, OnDestroy {
   /** MatButtonToggleGroup reads this to assign its own value. */
   @Input() value: any;
 
-  /** Tabindex for the toggle. */
-  @Input() tabIndex: number | null;
+  /** The tabindex of the button. */
+  @Input()
+  get tabIndex(): number | null {
+    return this._tabIndex;
+  }
+  set tabIndex(value: number | null) {
+    this._tabIndex = value;
+    this._changeDetectorRef.markForCheck();
```

Comment:
Consider using `MatButtonToggleGroup#_markButtonsForCheck`. That seems like it may be the established pattern for running change detection.

Reply:
Adding the `MatButtonToggleGroup#_markButtonsForCheck` call when we initialize the tab index turns out to be not sufficient. Angular will still ignore updates to the tab index when the tab index of a button toggle changes due to selection. To fix this, we can either add the `MatButtonToggleGroup#_markButtonsForCheck` call in every place where the tab index is updated, or we can call `MatButtonToggle#_markForCheck`, which is also an established pattern to handle change detection (It is an existing function in MatButtonToggle, just like _markButtonsForCheck). 

Two major advantages of using `MatButtonToggle#_markForCheck` are that we don’t need to have this call in multiple places and we can skip running change detection for the button when it is not necessary.